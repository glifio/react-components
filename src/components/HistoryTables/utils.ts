import { FilecoinNumber, BigNumber } from '@glif/filecoin-number'
import { MessageConfirmed } from '../../generated/graphql'
import makeFriendlyBalance from '../../utils/makeFriendlyBalance'

export function attoFilToFil(amount: string | number | BigNumber): string {
  return new FilecoinNumber(amount, 'attofil').toFil() + ' FIL'
}

export function getTotalCost(
  message: Pick<
    MessageConfirmed,
    'baseFeeBurn' | 'overEstimationBurn' | 'minerTip'
  >
): string {
  const bnBaseFeeBurn = new BigNumber(message.baseFeeBurn)
  const bnOverEstimationBurn = new BigNumber(message.overEstimationBurn)
  const bnMinerTip = new BigNumber(message.minerTip)
  const totalCost = bnBaseFeeBurn.plus(bnOverEstimationBurn).plus(bnMinerTip)
  return new FilecoinNumber(totalCost, 'attofil').toFil() + ' FIL'
}

export function getTotalCostShort(
  message: Pick<
    MessageConfirmed,
    'baseFeeBurn' | 'overEstimationBurn' | 'minerTip'
  >
): string {
  const bnBaseFeeBurn = new BigNumber(message.baseFeeBurn)
  const bnOverEstimationBurn = new BigNumber(message.overEstimationBurn)
  const bnMinerTip = new BigNumber(message.minerTip)
  const totalCost = bnBaseFeeBurn.plus(bnOverEstimationBurn).plus(bnMinerTip)
  const totalFil = new FilecoinNumber(totalCost, 'attofil')
  return makeFriendlyBalance(totalFil) + ' FIL'
}

export function getGasPercentage(
  message: Pick<MessageConfirmed, 'gasLimit' | 'gasUsed'>
): string {
  const gasLimit = new BigNumber(message.gasLimit)
  const gasUsed = new BigNumber(message.gasUsed)
  return gasUsed.dividedBy(gasLimit).times(100).toFixed(1) + '%'
}

export function formatNumber(input: string | number | BigNumber): string {
  const bigNr = input instanceof BigNumber ? input : new BigNumber(input)
  return bigNr.toFormat()
}
