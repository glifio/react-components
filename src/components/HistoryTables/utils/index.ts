import { FilecoinNumber, BigNumber } from '@glif/filecoin-number'
import { Message, MessagePending } from '../../../generated/graphql'

export function attoFilToFil(amount: string | number | BigNumber): string {
  return new FilecoinNumber(amount, 'attofil').toFil() + ' FIL'
}

export function getGasPercentage(
  message: Pick<Message, 'gasLimit' | 'gasCost'> | MessagePending,
  pending: boolean
): string {
  if (pending) return ''
  message = message as Message
  const gasLimit = new BigNumber(message.gasLimit)
  const gasUsed = new BigNumber(message.gasCost.gasUsed)
  return gasUsed.dividedBy(gasLimit).times(100).toFixed(1) + '%'
}

export function formatNumber(input: string | number | BigNumber): string {
  const bigNr = input instanceof BigNumber ? input : new BigNumber(input)
  return bigNr.toFormat()
}

export function removeMessageDups(
  existing: { __ref: any }[],
  incoming: { __ref: any }[],
  args: { offset: number; limit: number }
) {
  const head = [...existing]
  const tail = head.splice(args.offset * args.limit)
  const merged = [].concat(head, incoming, tail)

  const found = new Set([])
  return [...merged].filter(ele => {
    if (!found.has(ele.__ref)) {
      found.add(ele.__ref)
      return true
    }

    return false
  })
}
