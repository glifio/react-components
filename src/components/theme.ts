export const baseColors = {
  mono: {
    black: '#000',
    nearblack: '#262626',
    darkgray: '#666666',
    lightgray: '#C4C4C4',
    silver: '#999999',
    white: '#ffffff',
    transparent: 'transparent'
  },
  blue: {
    lightest: '#F6F8FE',
    lighter: '#EFF3FD',
    light: '#E4EBFC',
    mid: '#d1ddfa',
    mid2: '#bacbf7',
    muted: '#e8ecf7',
    muted700: '#d9e0f2'
  },
  green: {
    primary: '#1AD08F',
    light: '#D2F5ED',
    muted: '#85e0cb',
    dark: '#007056',
    darker: '#08442F'
  },
  red: {
    light: '#FC6D6F',
    dark: '#660000',
    base: '#ff4242'
  },
  orange: {
    light: '#FF9E80',
    dark: '#661800',
    mid: '#FFA826'
  },
  yellow: {
    light: '#FFDC99',
    deep: '#FCA703',
    mid: '#F7C103'
  },
  purple: {
    light: '#E0D7FE',
    mid: '#5E26FF',
    deep: '#290a85'
  }
}

// The core color object is the only non-semantic object here. This is because these colors are used so widely that it would be highly redundant to replicate these color values repeatedly within this "colors" object to style the text/background of every single Component the app renders.
const core = {
  primary: baseColors.purple.mid,
  secondary: baseColors.purple.light,
  tertiary: baseColors.purple.deep,
  black: baseColors.mono.black,
  nearblack: baseColors.mono.nearblack,
  darkgray: baseColors.mono.darkgray,
  lightgray: baseColors.mono.lightgray,
  silver: baseColors.mono.silver,
  white: baseColors.mono.white,
  transparent: baseColors.mono.transparent,
  yellow: baseColors.yellow.deep
}

const colors = {
  core,
  buttons: {
    primary: {
      background: baseColors.green.primary,
      borderColor: baseColors.green.primary,
      color: baseColors.green.darker
    },
    secondary: {
      background: baseColors.mono.transparent,
      borderColor: baseColors.mono.nearblack,
      color: baseColors.mono.nearblack
    },
    tertiary: {
      background: baseColors.mono.transparent,
      borderColor: baseColors.purple.light,
      color: baseColors.purple.light
    },
    'tertiary-selected': {
      background: baseColors.purple.mid,
      borderColor: baseColors.purple.mid,
      color: baseColors.purple.light
    }
  },
  background: {
    app: baseColors.blue.lightest,
    screen: baseColors.blue.lighter,
    text: baseColors.blue.light,
    messageHistory: baseColors.blue.muted
  },
  card: {
    account: {
      background: core.primary,
      color: core.secondary
    },
    balance: {
      background: baseColors.mono.transparent,
      color: core.nearblack
    },
    confirmation: {
      background: baseColors.green.light,
      foreground: baseColors.green.dark
    },
    error: {
      background: baseColors.red.light,
      foreground: baseColors.red.dark
    }
  },
  input: {
    background: {
      base: baseColors.blue.mid,
      active: baseColors.blue.mid2,
      valid: baseColors.green.muted,
      invalid: baseColors.red.light
    },
    border: core.silver
  },
  status: {
    success: {
      background: baseColors.green.primary,
      foreground: baseColors.green.darker
    },
    pending: {
      background: baseColors.yellow.deep,
      foreground: baseColors.mono.darkgray
    },
    warning: {
      background: baseColors.orange.light,
      foreground: baseColors.orange.dark
    },
    fail: {
      background: baseColors.red.light,
      foreground: baseColors.red.dark
    },
    inactive: baseColors.mono.lightgray
  },
  // gonna start slowly migrating colors from their inheritance vars above
  blue: {
    muted700: '#d9e0f2'
  },
  gray: {
    light: '#C4C4C4',
    medium: '#949598',
    dark: '#666666'
  },
  green: {
    primary: '#1AD08F',
    light: '#D2F5ED',
    muted: '#85e0cb',
    medium: '#007B39',
    dark: '#007056',
    darker: '#08442F'
  },
  yellow: {
    light: '#FFDC99',
    medium: '#C59900',
    deep: '#FCA703',
    mid: '#F7C103'
  },
  purple: {
    light: '#E0D7FE',
    medium: '#5E26FF',
    deep: '#290a85'
  }
}

const theme = {
  colors,
  fontSizes: [
    '0rem',
    '1rem',
    '1.125rem',
    '1.25rem',
    '1.5rem',
    '2rem',
    '3rem',
    '5rem'
  ],
  fontWeights: [0, 400, 600, 900],
  letterSpacings: [0, 1, 2, 4, 8],
  lineHeights: {
    solid: 1,
    title: 1.2,
    copy: 1.4
  },
  textStyles: {
    header: {
      fontSize: 'calc(48px + (48 - 32) * (100vw - 360px) / (1440 - 360))',
      fontWeight: 400,
      margin: 0,
      lineHeight: 'title',
      fontFamily: 'RT-Alias-Grotesk'
    },
    bigTitle: {
      fontSize: 6,
      fontWeight: 700,
      margin: 0,
      lineHeight: 'solid',
      fontFamily: 'RT-Alias-Medium'
    },
    title: {
      fontSize: 4,
      fontWeight: 400,
      margin: 0,
      lineHeight: 'title',
      fontFamily: 'RT-Alias-Grotesk'
    },
    text: {
      fontSize: 2,
      fontColor: core.nearblack,
      fontWeight: 400,
      lineHeight: 'copy',
      fontFamily: 'RT-Alias-Grotesk'
    },
    label: {
      fontSize: 2,
      fontColor: core.darkgray,
      fontWeight: 400,
      lineHeight: 'solid',
      fontFamily: 'RT-Alias-Grotesk'
    },
    num: {
      m: {
        fontSize: 4,
        fontWeight: 400,
        margin: 0,
        lineHeight: 'title',
        fontFamily: 'RT-Alias-Grotesk'
      },
      l: {
        fontSize: 5,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'title',
        fontFamily: 'RT-Alias-Grotesk'
      },
      xl: {
        fontSize: 6,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'solid',
        fontFamily: 'RT-Alias-Grotesk'
      },
      xxl: {
        fontSize: 7,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'solid',
        fontFamily: 'RT-Alias-Grotesk'
      }
    }
  },
  fonts: {
    AliasMedium:
      '"RT-Alias-Medium", "system-ui", "Segoe UI", "Roboto", Helvetica',
    AliasGrotesk:
      '"RT-Alias-Grotesk", "system-ui", "Segoe UI", "Roboto", Helvetica',
    sansSerif:
      '"system-ui", "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";',
    mono: '"system-ui", "Segoe UI", Roboto Mono, Helvetica, Arial, monospace, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";'
  },
  space: [0, 4, 8, 16, 24, 32, 48, 64, 128, 256],
  sizes: [
    0, 4, 8, 16, 24, 32, 48, 64, 80, 160, 240, 300, 480, 560, 640, 720, 800,
    960, 1024, 1200, 1440
  ],
  radii: ['0', '1px', '4px', '8px', '16px', '32px', '100px'],
  borders: [0, `1px solid`, `4px solid`],
  borderStyles: ['solid'],
  borderWidths: ['0', '1px', '2px', '4px'],
  breakpoints: ['40em', '52em', '64em'],
  shadows: [
    '0',
    '0 0.7px 2.2px -8px rgba(0, 0, 0, 0.18),0 1.7px 5.4px  rgba(0, 0, 0, 0.129),0 3.1px 10.1px rgba(0, 0, 0, 0.107),0 5.6px 18.1px rgba(0, 0, 0, 0.09),0 10.4px 33.8px rgba(0, 0, 0, 0.073),0 25px 81px rgba(0, 0, 0, 0.051)',
    'rgba(0, 0, 0, 0.10) 0px 0.7px 2.2px -8px, rgba(0, 0, 0, 0.04) 0px 1.7px 2.4px, rgba(0, 0, 0, 0.106) 0px 3.1px 8.1px, rgba(0, 0, 0, 0.04) 0px 5.6px 12.1px, rgba(0, 0, 0, 0.045) 0px 4.4px 4.8px, rgba(0, 0, 0, 0.05) 0px 15px 41px'
  ],
  opacity: {
    disabled: 0.4
  },
  zIndices: [0, 9, 99, 999, 9999]
}

export default theme

/* EVERYTHING BELOW HERE IS THEMEV2 */

// safe to change
const deviceWidthPxs = {
  phone: 480,
  tablet: 868,
  desktop: 1124
}

const spaces = {
  phone: {
    sm: '',
    small: '',
    default: '8px',
    md: '8px',
    medium: '8px',
    lg: '20px',
    large: '20px'
  },
  default: {
    sm: '',
    small: '',
    default: '10px',
    md: '10px',
    medium: '10px',
    lg: '20px',
    large: '20px'
  },
  tablet: {
    sm: '',
    small: '',
    default: '10px',
    md: '10px',
    medium: '10px',
    lg: '20px',
    large: '20px'
  },
  desktop: {
    sm: '',
    small: '',
    default: '10px',
    md: '10px',
    medium: '10px',
    lg: '20px',
    large: '20px'
  }
}

const fontSizes = {
  phone: {
    sm: '13px',
    small: '13px',
    default: '16px',
    md: '18px',
    medium: '18px',
    lg: '20px',
    large: '20px'
  },
  default: {
    sm: '13px',
    small: '13px',
    default: '18px',
    md: '26px',
    medium: '26px',
    lg: '28px',
    large: '28px'
  },
  tablet: {
    sm: '13px',
    small: '13px',
    default: '18px',
    md: '26px',
    medium: '26px',
    lg: '28px',
    large: '28px'
  },
  desktop: {
    sm: '13px',
    small: '13px',
    default: '18px',
    md: '26px',
    medium: '26px',
    lg: '28px',
    large: '28px'
  }
}

export const maxWidth = '1450px'

// dont change
export const devices = {
  phone: `${deviceWidthPxs.phone}px`,
  tablet: `${deviceWidthPxs.tablet}px`,
  desktop: `${deviceWidthPxs.desktop}px`
}

// these are
// one of size = sm, default, md, lg
// one of device = phone, tablet, desktop
export function space(size = 'default', device = 'default') {
  return spaces[device][size]
}

export function fontSize(size = 'default', device = 'default') {
  return fontSizes[device][size]
}
