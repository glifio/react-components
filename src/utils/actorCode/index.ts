import base32Decode from 'base32-decode'

export function decodeActorCID(cid: string): string {
  try {
    const decoded = base32Decode(cid.slice(1).toUpperCase(), 'RFC4648')
    const actorCode = new TextDecoder('utf-8').decode(decoded.slice(4))
    if (!actorCode.includes('fil/')) {
      return actorCode
    }
    return 'Unknown actor'
  } catch {
    return 'Unknown actor'
  }
}
