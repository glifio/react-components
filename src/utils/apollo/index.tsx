import { ReactNode } from 'react'
import {
  ApolloClient,
  ApolloProvider,
  InMemoryCache,
  HttpLink,
  split
} from '@apollo/client'
import { getMainDefinition } from '@apollo/client/utilities'
import { WebSocketLink } from '@apollo/client/link/ws'
import { defaultMessageHistoryClientCacheConfig } from './defaultCacheConfig'

import { useEnvironment } from '../../services'

const httpLink = (uri: string, secure: boolean): HttpLink =>
  new HttpLink({
    uri: `http${secure ? 's' : ''}://${uri}`
  })

const wsLink = (uri: string, secure: boolean): WebSocketLink =>
  new WebSocketLink({
    uri: `ws${secure ? 's' : ''}://${uri}`,
    options: {
      reconnect: true,
      lazy: true
    }
  })

export function createApolloClient(uri: string, secure: boolean) {
  // The split function takes three parameters:
  //
  // * A function that's called for each operation to execute
  // * The Link to use for an operation if the function returns a "truthy" value
  // * The Link to use for an operation if the function returns a "falsy" value

  // Only create the split in the browser
  const isBrowser = typeof window !== 'undefined'
  const link = isBrowser
    ? split(
        ({ query }) => {
          const definition = getMainDefinition(query)
          return (
            definition.kind === 'OperationDefinition' &&
            definition.operation === 'subscription'
          )
        },
        wsLink(uri, secure),
        httpLink(uri, secure)
      )
    : httpLink(uri, secure)

  return new ApolloClient({
    link,
    cache: new InMemoryCache({ ...defaultMessageHistoryClientCacheConfig })
  })
}

export function ApolloWrapper({ children }: { children: ReactNode }) {
  const { graphUrl, graphSecure } = useEnvironment()
  const client = createApolloClient(graphUrl, graphSecure)
  return <ApolloProvider client={client}>{children}</ApolloProvider>
}

export * from './defaultCacheConfig'
