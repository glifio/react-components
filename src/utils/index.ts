export { default as isMobileOrTablet } from './isMobileOrTablet'
export { default as useDesktopBrowser } from './useDesktopBrowser'
export { default as useChromeDesktopBrowser } from './useChromeDesktopBrowser'
export * from './useGetMessage'
export * from './useStateReadState'
export * from './actorCode'
export * from './isAddrEqual'
