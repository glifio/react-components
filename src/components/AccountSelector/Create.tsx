import { CoinType } from '@glif/filecoin-address'
import { useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { createPath } from '../../utils'
import { coinTypeCode } from '../../utils/createPath'
import { ButtonV2 } from '../Button/V2'
import { InputV2 } from '../InputV2'
import { Toggle } from '../InputV2/Toggle'
import { ShadowBox } from '../Layout'
import { space } from '../theme'

const CreateAccountContainer = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
  margin-top: ${space()};

  align-items: center;

  > * {
    width: fit-content;

    label {
      margin-top: ${space()};
    }

    button {
      width: 38.2%;
      min-width: fit-content;
    }
  }
`

const FormWrapper = styled(ShadowBox)`
  margin-top: ${space()};
`

const Path = styled.p`
  color: var(--gray-medium);
  padding-top: ${space()};
`

/**
 * TODO The label child overflow is a hack to get the UI to look right
 * but what really needs to happen is unit-wrapper class should have:
 * `width: fit-content`
 *
 * cant figure out how to get there, and this works pretty well..
 */
const Form = styled.form`
  > * {
    margin: ${space()};

    label {
      overflow: hidden;
    }
  }
`

enum CoinTypeOption {
  MAINNET = 'Mainnet',
  LEGACY = 'Legacy'
}

const coinTypeToCoinTypeOption = (ct: CoinType) => {
  if (ct === CoinType.TEST) {
    return CoinTypeOption.LEGACY
  }
  return CoinTypeOption.MAINNET
}

const coinTypeOptionToCoinType = (cto: CoinTypeOption) => {
  if (cto === CoinTypeOption.LEGACY) {
    return CoinType.TEST
  }
  return CoinType.MAIN
}

export function CreateAccount({
  fetchNextAccount,
  accountIdx: defaultAccountIdx
}: CreateAccountProps) {
  const [expertMode, setExpertMode] = useState(false)
  const [coinTypeOpt, setCoinTypeOpt] = useState<CoinTypeOption>(
    coinTypeToCoinTypeOption(process.env.NEXT_PUBLIC_COIN_TYPE as CoinType)
  )
  const [accountIndex, setAccountIndex] = useState(defaultAccountIdx)

  const path = isNaN(accountIndex)
    ? ''
    : createPath(
        coinTypeCode(coinTypeOptionToCoinType(coinTypeOpt)),
        accountIndex
      )

  return (
    <CreateAccountContainer>
      <ButtonV2
        large
        onClick={() =>
          fetchNextAccount(
            defaultAccountIdx,
            process.env.NEXT_PUBLIC_COIN_TYPE as CoinType
          )
        }
        disabled={expertMode}
      >
        Next account
      </ButtonV2>
      <Toggle
        label='Expert Mode'
        checked={expertMode}
        onChange={setExpertMode}
      />
      {expertMode && (
        <FormWrapper>
          <h3>Create account</h3>
          <Form
            onSubmit={e => {
              e.preventDefault()
              return fetchNextAccount(
                accountIndex,
                coinTypeOptionToCoinType(coinTypeOpt)
              )
            }}
          >
            <InputV2.Select
              label='Coin type'
              options={[CoinTypeOption.MAINNET, CoinTypeOption.LEGACY]}
              value={coinTypeOpt}
              onChange={v => setCoinTypeOpt(v as CoinTypeOption)}
            />
            <InputV2.Number
              label='Account index'
              value={accountIndex}
              onChange={setAccountIndex}
            />
            <Path>{path}</Path>
            <ButtonV2 type='submit'>Add account</ButtonV2>
          </Form>
        </FormWrapper>
      )}
    </CreateAccountContainer>
  )
}

type CreateAccountProps = {
  fetchNextAccount: (index: number, ct: CoinType) => void
  accountIdx: number
}

CreateAccount.propTypes = {
  fetchNextAccount: PropTypes.func.isRequired,
  accountIdx: PropTypes.number.isRequired
}
