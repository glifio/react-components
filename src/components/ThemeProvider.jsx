import PropTypes from 'prop-types'
import { ThemeProvider as StyledThemeProvider } from 'styled-components'
import { theme as defaultTheme } from './theme'

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  )
}

ThemeProvider.propTypes = {
  theme: PropTypes.object,
  children: PropTypes.node
}

ThemeProvider.defaultProps = {
  theme: defaultTheme,
  children: <></>
}

ThemeProvider.displayName = 'ThemeProvider'

export default ThemeProvider
