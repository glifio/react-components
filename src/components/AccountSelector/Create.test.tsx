import React from 'react'
import { act, render, screen, fireEvent, waitFor } from '@testing-library/react'
import { CoinType } from '@glif/filecoin-address'

import { CreateAccount } from './Create'
import ThemeProvider from '../ThemeProvider'
import theme from '../theme'
import { createPath } from '../../utils'
import { coinTypeCode } from '../../utils/createPath'

const spy = jest.fn().mockImplementation(() => Promise.resolve())

describe('Create new account', () => {
  test('it creates a new account at the wallet index by default', async () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <CreateAccount accountIdx={0} fetchNextAccount={spy} />
      </ThemeProvider>
    )
    expect(screen.getByText('Next account')).toBeInTheDocument()
    expect(screen.getByText('Expert Mode')).toBeInTheDocument()

    act(() => {
      fireEvent.click(screen.getByText('Next account'))
    })

    waitFor(() => {
      expect(spy).toHaveBeenCalledWith(0, CoinType.TEST)
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  test('it creates a new account at the specified index in expert mode', async () => {
    await act(async () => {
      render(
        <ThemeProvider theme={theme}>
          <CreateAccount accountIdx={0} fetchNextAccount={spy} />
        </ThemeProvider>
      )
      expect(screen.getByText('Next account')).toBeInTheDocument()
      expect(screen.getByText('Expert Mode')).toBeInTheDocument()

      const toggle = screen.getByRole('checkbox')
      toggle.focus()
      fireEvent.click(toggle)
      toggle.blur()

      await waitFor(() => {
        expect(screen.getByText('Create account')).toBeInTheDocument()
      })

      const accountIdx = screen.getByRole('spinbutton')
      accountIdx.focus()
      fireEvent.change(accountIdx, { target: { value: 10 } })
      accountIdx.blur()

      await waitFor(() => {
        expect(
          screen.getByText(createPath(coinTypeCode(CoinType.TEST), 10))
        ).toBeInTheDocument()
      })

      fireEvent.click(screen.getByText('Add account'))
      expect(spy).toHaveBeenCalledWith(10, CoinType.TEST)
    })
  })
})
