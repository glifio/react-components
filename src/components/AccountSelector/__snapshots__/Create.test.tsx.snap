// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create new account it creates a new account at the wallet index by default 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  gap: 0.75rem;
  padding: 0.5em 0.75em;
  color: var(--black);
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  border-color: var(--black);
  background: transparent;
  font-size: 1.25em;
}

.c1 svg path {
  stroke: var(--black);
}

.c1:hover {
  color: var(--purple-medium);
  border-color: var(--purple-medium);
}

.c1:hover svg path {
  stroke: var(--purple-medium);
}

.c1:active {
  color: var(--white) !important;
  border-color: var(--purple-medium) !important;
  background: var(--purple-medium) !important;
}

.c1:active svg path {
  stroke: var(--white) !important;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  text-align: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 > *:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 > *:last-child:not(.toggle-wrapper) {
  -webkit-flex: 0 0 50%;
  -ms-flex: 0 0 50%;
  flex: 0 0 50%;
}

.c2 span.info,
.c2 span.error {
  font-size: 0.875rem;
}

.c2 span.error {
  color: var(--red-medium);
}

.c3 .toggle-wrapper {
  position: relative;
}

.c3 input {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 0;
  height: 0;
  opacity: 0;
}

.c3 .toggle {
  display: inline-block;
  position: relative;
  width: 2.25em;
  height: 1.5em;
  border-radius: 0.75em;
  border: 2px solid var(--blue-medium);
  background-color: var(--blue-medium);
}

.c3 .toggle::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: calc(1.5em - 4px);
  height: calc(1.5em - 4px);
  border-radius: 50%;
  background-color: var(--white);
}

.c3 input:checked ~ .toggle {
  border-color: var(--blue-dark);
  background-color: var(--blue-dark);
}

.c3 input:checked ~ .toggle::after {
  left: auto;
  right: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 10px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > * {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0 > * label {
  margin-top: 10px;
}

.c0 > * button {
  width: 38.2%;
  min-width: -webkit-fit-content;
  min-width: -moz-fit-content;
  min-width: fit-content;
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    Next account
  </button>
  <label
    class="c2 c3"
  >
    <div>
      <span>
        Expert Mode
      </span>
      
    </div>
    <div
      class="toggle-wrapper"
    >
      <input
        type="checkbox"
      />
      <span
        class="toggle"
      />
    </div>
  </label>
</div>
`;
